### PR TITLE
Limit number of Threads used by Leshan client by default.

### PR DESCRIPTION
**Pull request by sbernard31**

### CODE REVIEW

File: leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java [hunk 1]
```diff
@@ -123,6 +123,16 @@ public LeshanClientBuilder setDtlsConfig(DtlsConnectorConfig.Builder config) {
         return this;
     }
 
+    public static NetworkConfig createDefaultNetworkConfig() {
+        NetworkConfig networkConfig = new NetworkConfig();
+        networkConfig.set(Keys.MID_TRACKER, "NULL");
+        networkConfig.set(Keys.MAX_ACTIVE_PEERS, 10);
+        networkConfig.set(Keys.MAX_MESSAGE_SIZE, 4096);
```
Comment (msangoi) [inline leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java]: MAX_RESOURCE_BODY_SIZE instead?
I don't know what is a "good" default value, so I would use the Californium default one.

File: leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java [hunk 2]
```diff
@@ -123,6 +127,16 @@ public LeshanClientBuilder setDtlsConfig(DtlsConnectorConfig.Builder config) {
         return this;
     }
 
+    public LeshanClientBuilder noUnsecureEndpoint() {
```
Comment (msangoi) [inline leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java]: disableSecureEndpoint() and disableUnsecureEndpoint()?
I think some Javadoc could be helpful here, to explain that localAddress/dtlsConfig may be ignored if the endpoints are disabled.

File: leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java [hunk 3]
```diff
@@ -236,6 +237,17 @@ public static void createAndStartClient(String endpoint, String localAddress, in
         List<LwM2mObjectEnabler> enablers = initializer.create(SECURITY, SERVER, DEVICE, LOCATION,
                 OBJECT_ID_TEMPERATURE_SENSOR);
 
+        // Create CoAP Config
+        NetworkConfig coapConfig;
```
Comment (msangoi) [inline leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java]: I think you should pass this coapConfig to the builder :)

**Comment by sbernard31** [comment]
Integrated in master (commit 0a824be1d3bc8cf512e5acf3597a57f6bb505824,  c728b0f69925bf40eaaf9c3c1280d586a10e55f5, e3479f84dfc38f0b9e15e477863b15b0a70f5c34, 1eb11893abf7f61880d45d38d875f1bbffedbd54)

**Pull request closed without merging**